Mage.Sets/src/mage/cards/a/AltarOfTheGoyf.java

```diff
@@ -0,0 +1,98 @@
+package mage.cards.a;
+
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.dynamicvalue.common.CardTypesInGraveyardCount;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.common.continuous.BoostTargetEffect;
+import mage.abilities.effects.common.continuous.GainAbilityControlledEffect;
+import mage.abilities.keyword.TrampleAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Duration;
+import mage.constants.SubType;
+import mage.constants.Zone;
+import mage.filter.FilterPermanent;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.target.targetpointer.FixedTarget;
+
+import java.util.UUID;
+
+/**
+ * @author jmharmon
+ */
+
+public final class AltarOfTheGoyf extends CardImpl {
+
+    private static final FilterPermanent filter = new FilterPermanent("Lhurgoyf creatures");
+
+    static {
+        filter.add(CardType.CREATURE.getPredicate());
+        filter.add(SubType.LHURGOYF.getPredicate());
+    }
+
+    public AltarOfTheGoyf(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.TRIBAL, CardType.ARTIFACT}, "{5}");
+
+        this.subtype.add(SubType.LHURGOYF);
+
+        // Whenever a creature you control attacks alone, it gets +X/+X until end of turn, where X is the number of card types among cards in all graveyard.
+        this.addAbility(new AltarOfTheGoyfAbility());
+
+        // Lhurgoyf creatures you control have trample.
+        this.addAbility(new SimpleStaticAbility(Zone.BATTLEFIELD, new GainAbilityControlledEffect(TrampleAbility.getInstance(), Duration.WhileOnBattlefield, filter)));
+    }
+
+    private AltarOfTheGoyf(final AltarOfTheGoyf card) {
+        super(card);
+    }
+
+    @Override
+    public AltarOfTheGoyf copy() {
+        return new AltarOfTheGoyf(this);
+    }
+}
+
+class AltarOfTheGoyfAbility extends TriggeredAbilityImpl{
+
+    public AltarOfTheGoyfAbility() {
+        super(Zone.BATTLEFIELD, new BoostTargetEffect(
+                CardTypesInGraveyardCount.instance, CardTypesInGraveyardCount.instance, Duration.EndOfTurn, true), false);
```

Comment:
this needs to count the card types in all graveyards, not just the controller's

Reply:
Done

Reply:
I'm stuck on this one.

Reply:
where are you stuck with it?

Reply:
Getting the cards types in all graveyards.

Reply:
you just need to iterate through all players, use `game.getState().getPlayersInRange()`

Reply:
The only place that seems to fit is the `if (game.isActivePlayer(this.controllerId))` section, but that doesn't work. Trying to look at the code for something like Tarmogoyf doesn't help either.

Reply:
You need to replace `CardTypesInGraveyardCount.instance` with your own class/enum that implements `DynamicValue`.  Put the code that iterates through the players in the `calculate` method.  Look at the code for `CardTypesInGraveyardCount` as an example to get you started.